### ISSUE TITLE
[BUG] Update location fragment on smooth scroll anchor link clicks

### ISSUE THREAD

**Issue by jrysana** [noodle-run/noodle#202]
**Describe the bug**
Per #201, smooth scrolling anchor links on the landing page should actually update the fragment in the URL bar when you click them on the landing page. Currently, they don't, because we prevent the default behavior of the `Link` (anchor tag wrapper) when on the same page to allow JS to smoothly scroll reliably to the anchor.

**To Reproduce**
- Try clicking around the navbar on the landing page

**Expected behavior**
- Browser URL bar fragment should update to the corresponding `id` when you click on the navbar links

**Additional context**
- It might make sense to do a custom wrapper around Next's `Link` component to facilitate automatic smooth scrolling on same-page anchor links.

**Comment by swapn652** [comment]
Hey, can I work on this?

**Comment by ajitzero** [comment]
I just noticed this issue myself and came to raise it. Here's a CSS-only fix that works in [most browsers](https://caniuse.com/css-scroll-behavior) for whoever picks it up:

- Update the HTML tag to use smooth scrolling via CSS.
  - Recommended: Update in global styles: `html { @apply smooth-scroll; }`
  - OR: `<html class="smooth-scroll">` but in the Next `Html` equivalent.
- Update the Link component to disable Next's scroll default behavior: `<Link scroll={false} {...otherProps}>`

Here's a related StackOverflow question for the history behind this issue: [https://stackoverflow.com/a/74084483](https://stackoverflow.com/a/74084483/8080186)

Unfortunately, I won't be able to help this week as I've got a lot on my plate. If nobody else volunteers, I am happy to take this up in the coming week.
